src/linear_combination/mod.rs

```diff
@@ -1,12 +1,37 @@
+//! 
 use crate::errors::PermError;
 use crate::mds_matrix::MDS_MATRIX;
 use crate::round_constants::ROUND_CONSTANTS;
 use bulletproofs::r1cs::{ConstraintSystem, LinearCombination};
 use curve25519_dalek::scalar::Scalar;
 
+/// Total ammount of full rounds that will be applied. 
+/// This is expressed as `RF` in the Poseidon paper.
 const TOTAL_FULL_ROUNDS: usize = 8;
+
+/// Total ammount of partial rounds that will be applied. 
+/// This is expressed as `Rp` in the Poseidon paper.
 const PARTIAL_ROUNDS: usize = 59;
 
+/// Applies a `permutation-round` of the `Poseidon252` hashing algorithm. 
+/// 
+/// It returns a vec of `WIDTH` outputs as a result which should be 
+/// a randomly permuted version of the input.  
+/// 
+/// In general, the same round function is iterated enough times
+/// to make sure that any symmetries and structural properties that
+/// might exist in the round function vanish.
+/// 
+/// This `permutation` is a 3-step process that:
+/// 
+/// - Applies twice the half of the `FULL_ROUNDS` 
+/// (which can be understood as linear ops).
+///  
+/// - In the middle step it applies the `PARTIAL_ROUNDS` 
+/// (which can be understood as non-linear ops).
+/// 
+/// This structure allows to minimize the number of non-linear
```

Comment:
Since we deleted L28 and L31, this sentence does not make much sense. We can probably delete this too